strata.js

```diff
@@ -809,17 +809,81 @@
+            }else{
+                solrQueryString = solrQueryString.concat(" AND " + param);
+            }
+        };        
+        if (!isObjectNothing(caseStatus)) {
+            identifier = '+case_status:';
+            if (caseStatus.toLowerCase() === 'open') {
+                concatQueryString(identifier + 'Waiting*');
+            } else if (caseStatus.toLowerCase() === 'closed') {
+                concatQueryString(identifier + 'Closed*');
+            } else{
+                concatQueryString(identifier + '*');
+            }
+        }
+        if (!isObjectNothing(caseOwner)) { 
+            identifier = '+case_owner:';
+            concatQueryString(identifier + caseOwner);
+        }
+        if (!isObjectNothing(caseGroup)) {
+            identifier = '+case_folderNumber:';
+            if (caseGroup === 'ungrouped') {
+                concatQueryString('+case_hasGroup:false');
+            } else {
+                concatQueryString(identifier + caseGroup);
+            }
+        }
+        if (!isObjectNothing(searchString)) {
+            identifier = 'allText:';
+            concatQueryString(identifier + searchString);
+        }
+        if (!isObjectNothing(queryParams) && queryParams.length > 0){
+            for (var i = 0; i < queryParams.length; ++i) {
+                concatQueryString(queryParams[i]);
+            }
+        }
+        if (!isObjectNothing(sortField)) {
+            identifier = '&sort=case_';
+            solrQueryString = solrQueryString.concat(identifier + sortField);
+        }
+        if (!isObjectNothing(sortOrder)) {
+            solrQueryString = solrQueryString.concat(" " + sortOrder);
+        }
+        if (!isObjectNothing(offset)) {
+            solrQueryString = solrQueryString.concat("&offset=" + offset);
+        }
+        if (!isObjectNothing(limit)) {
+            solrQueryString = solrQueryString.concat("&limit=" + limit);
+        }
+        return solrQueryString;
+    }
+
+    //Search cases SOLR
+    strata.cases.search = function (onSuccess, onFailure, caseStatus, caseOwner, caseGroup, searchString, sortField, sortOrder, offset, limit, queryParams) {
         if (!$.isFunction(onSuccess)) { throw 'onSuccess callback must be a function'; }
         if (!$.isFunction(onFailure)) { throw 'onFailure callback must be a function'; }
-        if (searchQuery === undefined) { onFailure('search query must be defined'); }
-
+        var searchQuery = prepareSolrQuery(caseStatus, caseOwner, caseGroup, searchString, sortField, sortOrder, offset, limit, queryParams);
+        
         var url = strataHostname.clone().setPath('/rs/cases');
         url.addQueryParam('query', searchQuery);
-        if (isSolrSearch) {
-            url.addQueryParam('newSearch', true);  // Add this query param to direct search to Calaveras
-        }
+        url.addQueryParam('newSearch', true);  // Add this query param to direct search to Calaveras
+        
```

Comment:
Good thinking on this @ardasgup.  Forcing all users to use Solr is the right way to go since the flag will be disappearing in the near future.  However this is a breaking change and when merged we should bump the minor version.